BigQuery/tests/System/ManageTablesTest.php

```diff
@@ -255,6 +255,79 @@ public function testIam()
         $this->assertEquals([$perm], $iam->testPermissions([$perm]));
     }
 
+    public function testSnapshot()
+    {
+        $destinationTable = self::$dataset->table(uniqid('test_dest_table_'));
+        $copyConfig = self::$table->copy(
+            $destinationTable,
+            ['configuration' => ['copy' => ['operationType' => 'SNAPSHOT']]],
+        );
+        $this->runJob($copyConfig);
+        $snapshotDefinition = $destinationTable->reload()['snapshotDefinition'];
+
+        // Assert for proper base table reference
+        $this->assertEquals(
+            $snapshotDefinition['baseTableReference'],
+            self::$table->info()['tableReference']
+        );
+
+        // Assert for proper RFC3339 extended format timestamp
+        $isRfc3339_extended = \DateTime::createFromFormat(
+            \DateTime::RFC3339_EXTENDED,
+            $snapshotDefinition['snapshotTime']
+        ) === false ? false : true;
+        $this->assertTrue($isRfc3339_extended);
+
+        // Test snapshot restore operation
+        $restoredTable = self::$dataset->table(uniqid('restored_table_'));
+        $copyConfig = $destinationTable->copy(
+            $restoredTable,
+            ['configuration' => ['copy' => ['operationType' => 'RESTORE']]],
+        );
+        $this->runJob($copyConfig);
+    }
+
+    public function testCreateClone()
+    {
+        $destinationTable = self::$dataset->table(uniqid('test_dest_table_'));
+        $copyConfig = self::$table->copy(
+            $destinationTable,
+            ['configuration' => ['copy' => ['operationType' => 'CLONE']]],
+        );
+        $this->runJob($copyConfig);
+        $cloneDefinition = $destinationTable->reload()['cloneDefinition'];
+
+        // Assert for proper base table reference
+        $this->assertEquals(
+            $cloneDefinition['baseTableReference'],
+            self::$table->info()['tableReference']
+        );
+
+        // Assert for proper RFC3339 extended format timestamp
+        $isRfc3339_extended = \DateTime::createFromFormat(
+            \DateTime::RFC3339_EXTENDED,
+            $cloneDefinition['cloneTime']
+        ) === false ? false : true;
+        $this->assertTrue($isRfc3339_extended);
+
+        return $destinationTable;
+    }
```

Comment:
The reason for me not using a common method and a `dataprovider` to test for `SNAPSHOT` and `CLONE` creation is that there are further specific tests for particular copies which would depend on snapshot/clone creation method.

<details>
 <summary>Code Example</summary>
 
```php
/**
* @dataProvider operationTypes()
*/
public function testCreateSnapshotAndClone(){}

/**
* The following methods requires the respective table instances which it
* should get from @depends annotation but not possible
*/
public function testRestoreSnapshot(){}

public function testUpdateClone(){}

public function operationTypes()
{
    return [
        'SNAPSHOT`,
        'CLONE`
    ];
}
```
</details>

Reply:
Did you consider this:

Have individual data providers for each case. Use multiple dataproviders on one test method and individual data providers on other test methods.